compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -982,7 +982,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
         }
     }
 
-    fn assoc_ty_contraint_param_error_emit(&self, data: &ParenthesizedArgs) -> () {
+    fn assoc_ty_constraint_param_error_emit(&self, data: &ParenthesizedArgs) -> () {
```

Comment:
Could you rename this to something starting with `emit_`? Perhaps `emit_bad_parenthesized_arg_in_assoc_ty`? It's more consistent with other functions that emit errors.

Also, please remove the `-> ()`, because those are redundant.